sympy/solvers/tests/test_ode.py

```diff
@@ -3705,3 +3705,29 @@ def test_issue_17322():
     sol = [Eq(f(x), C1), Eq(f(x), C1*exp(-x))]
     assert set(sol) == set(dsolve(eq, hint='lie_group'))
     assert checkodesol(eq, sol) == 2*[(True, 0)]
+
+def test_2nd_2F1_hypergeometric():
+
+    a, b, c = symbols("a b c")
+
+    eq = x*(x-1)*f(x).diff(x, 2) + ((a+b+1)*x-c)*f(x).diff(x) + a*b*f(x)
+    sol = Eq(f(x), C1*x**(1 - c)*hyper((b - c + 1, a - c + 1), (2 - c,), x) + C2*hyper((b, a), (c,), x))
+    assert sol == dsolve(eq, hint='2nd_hypergeometirc')
+    # assert checkodesol(eq, sol) == (True, 0) Hanging
+
+    eq = x*(x-1)*f(x).diff(x, 2) + (S(3)/2 -2*x)*f(x).diff(x) + 2*f(x)
+    sol = Eq(f(x), C1*x**(S(5)/2)*hyper((S(1)/2, S(3)/2), (S(7)/2,), x) + C2*hyper((-2, -1), (-S(3)/2,), x))
+    assert sol == dsolve(eq, hint='2nd_hypergeometirc')
+    assert checkodesol(eq, sol) == (True, 0)
+
+    eq=x*(x-1)*f(x).diff(x, 2) + (S(3) -2*x)*f(x).diff(x) + 2*f(x)
+    sol = Eq(f(x), C1*x**4*hyper((2, 3), (5,), x) + C2*hyper((-2, -1), (-3,), x))
+    assert sol == dsolve(eq, hint='2nd_hypergeometirc')
+    assert checkodesol(eq, sol) == (True, 0)
+
+    eq = -x**(S(5)/7)*(-416*x**(S(9)/7)/9 - 2385*x**(S(5)/7)/49 + S(298)*x/3)*f(x)/(196*(-x**(S(6)/7) +
+         x)**2*(x**(S(6)/7) + x)**2) + Derivative(f(x), (x, 2))
+    sol = Eq(f(x), x**(S(45)/98)*(C1*x**(S(4)/49)*hyper((S(-1)/2, S(1)/3), (S(9)/7,), x**(S(2)/7)) +
+          C2*hyper((S(-11)/14, S(1)/21), (S(5)/7,), x**(S(2)/7)))/(x**(S(2)/7) - 1)**(S(19)/84))
+    assert sol == dsolve(eq, hint='2nd_hypergeometirc')
+    # assert checkodesol(eq, sol) == (True, 0) Hanging
```

Comment:
What exactly is hanging? Can we open an issue for that?

Reply:
I checked it again and It is taking time. Also not able to simplify the expression and returning `False`. But the solution is correct and I have checked it from maple.

Reply:
What is the expression that should simplify? We can use that to open an issue for simplify.

Reply:
`12005*C1*x**(256/49)*hyper((3/2, 7/3), (23/7,), x**(2/7)) + 34300*C1*x**(242/49)*hyper((1/2, 4/3), (16/7,), x**(2/7)) - 48020*C1*x**(242/49)*hyper((3/2, 7/3), (23/7,), x**(2/7)) + 52920*C1*x**(228/49)*hyper((-1/2, 1/3), (9/7,), x**(2/7)) - 155820*C1*x**(228/49)*hyper((1/2, 4/3), (16/7,), x**(2/7)) + 72030*C1*x**(228/49)*hyper((3/2, 7/3), (23/7,), x**(2/7)) - 158760*C1*x**(214/49)*hyper((-1/2, 1/3), (9/7,), x**(2/7)) + 261660*C1*x**(214/49)*hyper((1/2, 4/3), (16/7,), x**(2/7)) - 48020*C1*x**(214/49)*hyper((3/2, 7/3), (23/7,), x**(2/7)) + 158760*C1*x**(200/49)*hyper((-1/2, 1/3), (9/7,), x**(2/7)) - 193060*C1*x**(200/49)*hyper((1/2, 4/3), (16/7,), x**(2/7)) + 12005*C1*x**(200/49)*hyper((3/2, 7/3), (23/7,), x**(2/7)) - 52920*C1*x**(186/49)*hyper((-1/2, 1/3), (9/7,), x**(2/7)) + 52920*C1*x**(186/49)*hyper((1/2, 4/3), (16/7,), x**(2/7)) + 2178*C2*x**(36/7)*hyper((17/14, 43/21), (19/7,), x**(2/7)) + 4356*C2*x**(34/7)*hyper((3/14, 22/21), (12/7,), x**(2/7)) - 8712*C2*x**(34/7)*hyper((17/14, 43/21), (19/7,), x**(2/7)) + 11880*C2*x**(32/7)*hyper((-11/14, 1/21), (5/7,), x**(2/7)) - 24948*C2*x**(32/7)*hyper((3/14, 22/21), (12/7,), x**(2/7)) + 13068*C2*x**(32/7)*hyper((17/14, 43/21), (19/7,), x**(2/7)) - 35640*C2*x**(30/7)*hyper((-11/14, 1/21), (5/7,), x**(2/7)) + 48708*C2*x**(30/7)*hyper((3/14, 22/21), (12/7,), x**(2/7)) - 8712*C2*x**(30/7)*hyper((17/14, 43/21), (19/7,), x**(2/7)) - 11880*C2*x**(26/7)*hyper((-11/14, 1/21), (5/7,), x**(2/7)) + 11880*C2*x**(26/7)*hyper((3/14, 22/21), (12/7,), x**(2/7)) + 35640*C2*x**4*hyper((-11/14, 1/21), (5/7,), x**(2/7)) - 39996*C2*x**4*hyper((3/14, 22/21), (12/7,), x**(2/7)) + 2178*C2*x**4*hyper((17/14, 43/21), (19/7,), x**(2/7))`

Reply:
I've created #17702 can you add a link to it in the code here